src/openrct2/interface/InteractiveConsole.cpp

```diff
@@ -1714,40 +1708,28 @@ void InteractiveConsole::Execute(const std::string& s)
 
         if (length > 0)
         {
-            utf8* arg = (utf8*)malloc(length + 1);
-            std::memcpy(arg, start, length);
-            arg[length] = 0;
-
-            if (argc >= argvCapacity)
-            {
-                argvCapacity *= 2;
-                argv = (utf8**)realloc(argv, argvCapacity * sizeof(utf8*));
-            }
-            argv[argc] = arg;
-            argc++;
+            argv.emplace_back(start, length);
         }
 
         start = end;
     } while (*end != 0);
 
-    if (argc == 0)
+    if (argv.empty())
         return;
 
     bool validCommand = false;
+
     for (const auto& c : console_command_table)
     {
-        if (strcmp(argv[0], c.command) == 0)
+        if (argv[0] == std::string(c.command))
```

Comment:
std::string construction is not required here, it can compare against const char*